include/jsonbuilder/JsonBuilder.h

```diff
@@ -36,44 +36,88 @@ JsonBuilder design decisions:
   8 bytes), floating-point (4 or 8 bytes), boolean (true/false), null, time
-  (FILETIME), UUID (GUID), and string (utf-16). The convenience methods and the
-  renderer can be extended if other types are needed.
+  (in 100ns units since 1601, i.e. Windows FILETIME), UUID (big-endian order),
+  and string (UTF-8). The convenience methods and the renderer can be extended
+  if other types are needed.
 - Complex nodes come in two types: Object and Array. The JsonBuilder itself
   makes no distinction between these two types (they have identical behavior)
   but it is intended that the Object type contain named values (i.e. it is a
   dictionary with string keys) and that the Array type contain anonymous values
   (i.e. it is a list).
-- Value name limited to 16M utf-16 characters per value.
+- Value name limited to 16M bytes (UTF-8) per value.
 - Value data limited to 3GB per value.
-- Names are stored as utf-16.
-- Memory usage for object and array values is (in bytes):
+- Names are stored as UTF-8.
+- Memory usage for Complex (object and array) values is (in bytes):
   20 + sizeof(name) + padding to a multiple of 4.
-- Memory usage for all other values is (in bytes):
+- Memory usage for Simple (all other) values is (in bytes):
   12 + sizeof(name) + sizeof(data) + padding to multiple of 4.
 - Total storage limited to 16GB per JsonBuilder (or available VA space).
 
 Error handling:
 
-    Throw std::exception variants
+- Some functions may throw std::bad_alloc for memory allocation failure or
+  std::length_error if a size limit is exceeded.
+- Some functions may assert for precondition violation.
+- JsonBuilder uses JsonThrowBadAlloc(), JsonThrowLengthError(), and
+  JsonThrowInvalidArgument() functions to raise exceptions. The
+  library-provided implementations simply throw the corresponding exception
+  (e.g. std::bad_alloc). If necessary, you can override these at link time in
+  your own project by providing implementations as follows:
+
+#include <jsonbuilder/JsonBuilder.h>
+namespace jsonbuilder
+{
+    [[noreturn]] void _jsonbuilderDecl JsonThrowBadAlloc() noexcept(false)
+    {
+        throw MyBadAllocException();
+    }
+    [[noreturn]] void _jsonbuilderDecl JsonThrowLengthError(const char* what) noexcept(false)
+    {
+        throw MyLengthErrorException(what);
+    }
+    [[noreturn]] void _jsonbuilderDecl JsonThrowInvalidArgument(const char* what) noexcept(false)
+    {
+        throw MyInvalidArgumentException(what);
+    }
+}
 */
 
 #pragma once
-#include <algorithm>  // std::min
-#include <chrono>
-#include <cstring>
-#include <inttypes.h>
-#include <math.h>     // isfinite, sqrt
-#include <stdexcept>  // std::invalid_argument
-#include <string_view>
-#include <string>
 
-#include <uuid/uuid.h>
+#include <chrono>       // std::chrono::system_clock::time_point
+#include <iterator>     // std::forward_iterator_tag
+#include <string_view>  // std::string_view
+#include <type_traits>  // std::decay
```

Comment:
Note: Reducing `#include`s to just what is actually needed. Importantly, this header doesn't need to depend on `uuid.h`.